ACE/ace/INET_Addr.cpp

```diff
@@ -152,11 +152,38 @@ ACE_INET_Addr::hash (void) const
   return this->get_ip_address () + this->get_port_number ();
 }
 
+bool
+ACE_INET_Addr::next (void)
+{
+  if (this->inet_addrs_.empty () ||
+      this->inet_addrs_iter_ == this->inet_addrs_.end ())
+    return false;
+
+  union ip46 next_a = *this->inet_addrs_iter_++;
+  this->set_addr (&next_a, sizeof (next_a));
+  return true;
+}
+
+void
+ACE_INET_Addr::reset (void)
+{
+  this->inet_addrs_iter_ = this->inet_addrs_.begin ();
+  this->next ();
+  return;
+}
+
 ACE_INET_Addr::ACE_INET_Addr (void)
   : ACE_Addr (determine_type (), sizeof (inet_addr_))
 {
   // ACE_TRACE ("ACE_INET_Addr::ACE_INET_Addr");
-  this->reset ();
+  this->reset_i ();
+}
+
+ACE_INET_Addr &
+ACE_INET_Addr::operator= (const ACE_INET_Addr& rhs)
```

Comment:
Shouldn't this check for this != rhs? Also please add a test case for self assignment